src/services/dependencies.service.js

```diff
@@ -1,73 +1,35 @@
 // @flow
 import { PACKAGE_MANAGER_CMD } from './platform.services';
+import { ChildProcess } from 'child_process';
 const childProcess = window.require('child_process');
 
-/* TODO: Improve flow type for executor - no type annotation found
-         from facebook flow source https://github.com/facebook/flow/blob/v0.76.0/lib/core.js#L584:
-          constructor(callback: (
-                resolve: (result: Promise<R> | R) => void,
-                reject: (error: any) => void
-              ) => mixed): void;
-*/
-
-const spawnProcess = (
-  cmd: string,
-  cmdArgs: Array<string>,
-  projectPath: string,
-  [resolve, reject]: Array<any>
-) => {
-  const child = childProcess.spawn(cmd, cmdArgs, {
-    cwd: projectPath,
-  });
-
-  child.on('exit', code => {
-    if (code) {
-      reject(child.stderr);
-    } else {
-      resolve(child.stdout);
-    }
+const spawnProcess = (cmd: string, cmdArgs: string[], projectPath: string) =>
```

Comment:
@superhawk610 what do you think should we move `spawnProcess` to another service and import it here? I think this could be useful as it uses promises and is reusable. If this is useful how should I name that service as I think there is no place for it at the moment. The service is maybe not needed if we would use `child-process-promise` see link below.

I haven't checked where it could be used but probably in `create-project.service`. We just need to add one additional parameter so we can add callbacks for event handlers e.g.` child.stdout.on('data', handler)`.

The api of the parameter could be `{stream: 'stdout', event: 'data', handler: () => {}}`  so we can register the handler on child like this `child[stream].on(event, handler)`. For event and stream I would use stdout & data as default so it's not needed to pass.
Also an array instead of the object could be passed to add multiple events - requires type check for object and array.
So I think it would be great to have the followng api `spawnProcess(cmd, cmdArgs, projectPath, object|array|function) : Promise`.
And I would make `projectPath` & `eventHandler` optional for use-cases where they're not required.

Or we could also use [child-process-promise](https://www.npmjs.com/package/child-process-promise) so we're not reinventing the wheel. (Found it after a quick search - not checked if it works for all use-cases.)